migration/api.py

```diff
@@ -14,22 +20,58 @@ class EndpointType(Enum):
     REST = '/api/v1/'
 
 
+@dataclass
+class GetResponseData:
+    data: Any
+    next_page_params: dict[str, Any] | None
+
+
 class API:
     client: httpx.Client
 
-    def __init__(self, url: str, key: str, endpoint_type: EndpointType = EndpointType.REST):
-        headers = { 'Authorization': f'Bearer {key}'}
-        self.client = httpx.Client(base_url=url + endpoint_type.value, headers=headers)
+    def __init__(
+        self,
+        url: str,
+        key: str, endpoint_type: EndpointType = EndpointType.REST,
+        timeout: int = 10
+    ):
+        headers = { 'Authorization': f'Bearer {key}' }
+        self.client = httpx.Client(base_url=url + endpoint_type.value, headers=headers, timeout=timeout)
 
-    def get_next_page_params(self, resp: httpx.Response) -> dict[str, Any] | None:
+    @staticmethod
+    def get_next_page_params(resp: httpx.Response) -> dict[str, Any] | None:
         if 'next' not in resp.links:
             return None
         else:
             query_params = parse_qs(urlparse(resp.links['next']['url']).query)
             return query_params
 
+    @retry(
+        stop=stop_after_attempt(4),
+        retry=retry_if_exception_type((httpx.HTTPError, JSONDecodeError)),
+        reraise=True,
+        before_sleep=before_sleep_log(logger, logging.WARN)
+    )
+    def get(self, url: str, params: dict[str, Any] | None = None) -> GetResponseData:
+        resp = self.client.get(url=url, params=params)
+        resp.raise_for_status()
+        data = resp.json()
+        next_page_params = self.get_next_page_params(resp)
+        return GetResponseData(data, next_page_params)
+
+    @retry(
+        stop=stop_after_attempt(4),
```

Comment:
I think it's probably possible to make this configurable, but having done so in the past with other projects, this doesn't seem like something we ever change. If we did, I think it'd just have to be a environment variable/global variable in this file, since there isn't a good way to pass class instance variables to decorators.

Reply:
We can leave the retry number hard coded for now. You can remove this item from the checklist.